internal/bminventory/inventory.go

```diff
@@ -6158,6 +6174,178 @@ func (b *bareMetalInventory) V2UpdateHostIgnitionInternal(ctx context.Context, p
+
+	osImage, err := b.osImages.GetOsImageOrLatest(infraEnv.OpenshiftVersion, infraEnv.CPUArchitecture)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get OS image for OVE ignition generation")
+	}
+
+	tempDir, err := os.MkdirTemp("", "ove-ignition-*")
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create temp directory for OVE ignition generation")
+	}
+	defer os.RemoveAll(tempDir)
+
+	manifestsDir := filepath.Join(tempDir, "cluster-manifests")
+	if err = os.MkdirAll(manifestsDir, 0755); err != nil {
+		return "", errors.Wrap(err, "failed to create manifests directory")
+	}
+
+	infraEnvObj := struct {
+		APIVersion string `yaml:"apiVersion"`
+		Kind       string `yaml:"kind"`
+		Metadata   struct {
+			Name string `yaml:"name"`
+		} `yaml:"metadata"`
+		Spec struct {
+			PullSecretRef *struct {
+				Name string `yaml:"name"`
+			} `yaml:"pullSecretRef,omitempty"`
+			SSHAuthorizedKey string `yaml:"sshAuthorizedKey,omitempty"`
+		} `yaml:"spec"`
+	}{
+		APIVersion: "agent-install.openshift.io/v1beta1",
+		Kind:       "InfraEnv",
+		Metadata: struct {
+			Name string `yaml:"name"`
+		}{
+			Name: *infraEnv.Name,
+		},
+		Spec: struct {
+			PullSecretRef *struct {
+				Name string `yaml:"name"`
+			} `yaml:"pullSecretRef,omitempty"`
+			SSHAuthorizedKey string `yaml:"sshAuthorizedKey,omitempty"`
+		}{
+			PullSecretRef: &struct {
+				Name string `yaml:"name"`
+			}{Name: "pull-secret"},
+			SSHAuthorizedKey: infraEnv.SSHAuthorizedKey,
+		},
+	}
+
+	infraEnvYAML, err := yaml.Marshal(infraEnvObj)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to marshal infraEnv YAML")
+	}
+	if err = os.WriteFile(filepath.Join(manifestsDir, "infraenv.yaml"), infraEnvYAML, 0600); err != nil {
+		return "", errors.Wrap(err, "failed to write infraenv.yaml")
+	}
+
+	pullSecretObj := struct {
+		APIVersion string `yaml:"apiVersion"`
+		Kind       string `yaml:"kind"`
+		Metadata   struct {
+			Name string `yaml:"name"`
+		} `yaml:"metadata"`
+		Type       string            `yaml:"type"`
+		StringData map[string]string `yaml:"stringData"`
+	}{
+		APIVersion: "v1",
+		Kind:       "Secret",
+		Type:       "kubernetes.io/dockerconfigjson",
+		StringData: map[string]string{
+			".dockerconfigjson": infraEnv.PullSecret,
+		},
+	}
+	pullSecretObj.Metadata.Name = "pull-secret"
+
+	pullSecretYAML, err := yaml.Marshal(pullSecretObj)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to marshal pull secret YAML")
+	}
+	if err = os.WriteFile(filepath.Join(manifestsDir, "pull-secret.yaml"), pullSecretYAML, 0600); err != nil {
+		return "", errors.Wrap(err, "failed to write pull-secret.yaml")
+	}
+
+	mirrorDir := filepath.Join(tempDir, "mirror")
+	if err = os.MkdirAll(mirrorDir, 0755); err != nil {
+		return "", errors.Wrap(err, "failed to create mirror directory")
+	}
+
+	if err = os.WriteFile(filepath.Join(mirrorDir, "registries.conf"), []byte(getOVERegistriesConf()), 0600); err != nil {
```

Comment:
Should this also include any registries the user configured in our APIs? Or any set up in the environment like we do for other cases?